client-web/src/exlibris-app/reducers/exlibris-reducer.html

```diff
@@ -39,9 +44,58 @@
                     //TODO:2018-55-17:yurii.haidamaka add data loading from the server
                     return state;
                 }
+                case ExlibrisApp.ExlibrisActions.FIND_BOOK_BY_ID: {
+                    const id = action.bookId;
+                    const book = findElement(id, state.allBooks, state.expectedSoonBooks, state.borrowedBooks, state.reservedBooks);
+                    if (book) {
+                        return Object.assign({}, state, {
+                            bookDetails: {
+                                book: book,
+                                loadingStatus: ExlibrisApp.LoadingStatus.COMPLETED
+                            }
+                        });
+                    }
+                    return Object.assign({}, state, {
+                        bookDetails: {
+                            book: {},
+                            loadingStatus: ExlibrisApp.LoadingStatus.NOT_FOUND
```

Comment:
Empty book + loaded:false  = Book not found.
Book + loaded:false  = Completed.
It's enough to have boolean property `loaded` instead of enum LoadingStatus.